sbank/overlap_cpu_lib.c

```diff
@@ -113,7 +113,7 @@ static WS *get_workspace(WS *workspace_cache, const size_t n) {
 }
 
 /* by default, complex arithmetic will call built-in function __muldc3, which does a lot of error checking for inf and nan; just do it manually */
-static void multiply_conjugate(float complex * restrict out, float complex *a, float complex *b, const size_t size) {
+static void multiply_conjugate(float complex * out, float complex *a, float complex *b, const size_t size) {
```

Comment:
This is not a compiler version issue (sort of), it's a c standard issue. `restrict` is part of c99, so you just need to make sure the compiler is using it:

```console
$ which gcc
/usr/bin/gcc
$ gcc --version
gcc (GCC) 4.8.5 20150623 (Red Hat 4.8.5-44)
Copyright (C) 2015 Free Software Foundation, Inc.
This is free software; see the source for copying conditions.  There is NO
warranty; not even for MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.

$ ./venv/bin/python3 setup.py build
running build
running build_py
creating build
creating build/lib.linux-x86_64-3.6
creating build/lib.linux-x86_64-3.6/sbank
copying sbank/__init__.py -> build/lib.linux-x86_64-3.6/sbank
copying sbank/tau0tau3.py -> build/lib.linux-x86_64-3.6/sbank
copying sbank/waveforms.py -> build/lib.linux-x86_64-3.6/sbank
copying sbank/psds.py -> build/lib.linux-x86_64-3.6/sbank
copying sbank/bank.py -> build/lib.linux-x86_64-3.6/sbank
copying sbank/overlap_cuda.py -> build/lib.linux-x86_64-3.6/sbank
copying sbank/overlap.py -> build/lib.linux-x86_64-3.6/sbank
creating build/lib.linux-x86_64-3.6/sbank/tests
copying sbank/tests/test_psds.py -> build/lib.linux-x86_64-3.6/sbank/tests
copying sbank/tests/__init__.py -> build/lib.linux-x86_64-3.6/sbank/tests
running build_ext
building 'sbank.overlap_cpu' extension
creating build/temp.linux-x86_64-3.6
creating build/temp.linux-x86_64-3.6/sbank
gcc -pthread -Wno-unused-result -Wsign-compare -DNDEBUG -O2 -g -pipe -Wall -Wp,-D_FORTIFY_SOURCE=2 -fexceptions -fstack-protector-strong --param=ssp-buffer-size=4 -grecord-gcc-switches -m64 -mtune=generic -D_GNU_SOURCE -fPIC -fwrapv -fPIC -Isbank -I/usr/lib64/python3.6/site-packages/numpy/core/include -I/home/duncan.macleod/git/sbank/venv/include -I/usr/include/python3.6m -c sbank/overlap_cpu.c -o build/temp.linux-x86_64-3.6/sbank/overlap_cpu.o -O3 -w -ffast-math -ffinite-math-only
In file included from sbank/overlap_cpu.c:643:0:
sbank/overlap_cpu_lib.c:116:57: error: expected ‘;’, ‘,’ or ‘)’ before ‘out’
 static void multiply_conjugate(float complex * restrict out, float complex *a, float complex *b, const size_t size) {
                                                         ^
error: command 'gcc' failed with exit status 1
$ CFLAGS="${CFLAGS} -std=c99" ./venv/bin/python3 setup.py build
running build
running build_py
running build_ext
building 'sbank.overlap_cpu' extension
gcc -pthread -DNDEBUG -O2 -g -pipe -Wall -Wp,-D_FORTIFY_SOURCE=2 -fexceptions -fstack-protector-strong --param=ssp-buffer-size=4 -grecord-gcc-switches -m64 -mtune=generic -D_GNU_SOURCE -fPIC -fwrapv -std=c99 -fPIC -Isbank -I/usr/lib64/python3.6/site-packages/numpy/core/include -I/home/duncan.macleod/git/sbank/venv/include -I/usr/include/python3.6m -c sbank/overlap_cpu.c -o build/temp.linux-x86_64-3.6/sbank/overlap_cpu.o -O3 -w -ffast-math -ffinite-math-only
gcc -pthread -shared -Wl,-z,relro -g -std=c99 build/temp.linux-x86_64-3.6/sbank/overlap_cpu.o -L/usr/lib64 -llal -lpython3.6m -o build/lib.linux-x86_64-3.6/sbank/overlap_cpu.cpython-36m-x86_64-linux-gnu.so
running build_scripts
creating build/scripts-3.6
copying and adjusting bin/sbank -> build/scripts-3.6
copying and adjusting bin/sbank_choose_mchirp_boundaries -> build/scripts-3.6
copying and adjusting bin/sbank_hdf5_bankcombiner -> build/scripts-3.6
copying and adjusting bin/sbank_hdf5_choose_mchirp_boundaries -> build/scripts-3.6
copying and adjusting bin/sbank_pipe -> build/scripts-3.6
copying and adjusting bin/sbank_plot_sim -> build/scripts-3.6
copying and adjusting bin/sbank_sim -> build/scripts-3.6
copying and adjusting bin/sbank_sim_pipe -> build/scripts-3.6
changing mode of build/scripts-3.6/sbank from 664 to 775
changing mode of build/scripts-3.6/sbank_choose_mchirp_boundaries from 664 to 775
changing mode of build/scripts-3.6/sbank_hdf5_bankcombiner from 664 to 775
changing mode of build/scripts-3.6/sbank_hdf5_choose_mchirp_boundaries from 664 to 775
changing mode of build/scripts-3.6/sbank_pipe from 664 to 775
changing mode of build/scripts-3.6/sbank_plot_sim from 664 to 775
changing mode of build/scripts-3.6/sbank_sim from 664 to 775
changing mode of build/scripts-3.6/sbank_sim_pipe from 664 to 775
```

Reply:
In other words, I don't think this change should be merged.

Reply:
Thanks for that. I guess I was knew that there was a proper solution to this (otherwise the code would never have compiled on CIT within lalsuite). I just figured that this probably won't make any difference if removed, and so the simple thing would be the most efficient fix.

However, given that you've explained how to solve this properly, I'll swap that in.